d365fo.tools/functions/send-d365broadcastmessage.ps1

```diff
@@ -111,27 +129,50 @@ function Send-D365BroadcastMessage {
         [datetime] $StartTime = (Get-Date),
 
         [Parameter(Mandatory = $false, Position = 7)]
-        [int] $EndingInMinutes = $Script:BroadcastEndingInMinutes
+        [int] $EndingInMinutes = $Script:BroadcastEndingInMinutes,
+
+        [Parameter(Mandatory = $false, Position = 8)]
+        [switch] $OnPremise = $Script:BroadcastOnPremise
     )
 
```

Comment:
Could be fixed with
$bearerParms = @{
Resource        = $URL
            ClientId        = $ClientId
            ClientSecret    = $ClientSecret
}
if ($OnPremise){
$bearerParms.AuthProviderUri = "$Tenant/oauth2/token"
} Else {
$bearerParms.AuthProviderUri = "https://login.microsoftonline.com/$Tenant/oauth2/token"
}

Reply:
less code you write, less code to maintain :) i fix it

Reply:
You are overwriting the variable :)

This should be the exact code:

```
if ($OnPremise) {
$bearerParms.AuthProviderUri = "$Tenant/oauth2/token"
} else{
$bearerParms.AuthProviderUri = "https://login.microsoftonline.com/$Tenant/oauth2/token"
}
```
Because you already created the $bearerParms with the 3 properties. Now we just need to add an extra property, which is done like I show here :)

Reply:
yes i see and fix it in my last commit

Reply:
+ fix trailing space